### PR TITLE
vmTools: misc debian tool improvements

### PR DESCRIPTION
**Pull request by lheckemann**
## Description of changes
These make the deb-based distro image generation tools more versatile.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by katexochen** [comment]
Do you have an example to test this?

**Comment by lheckemann** [comment]
Here's what I did using this version of the code: https://github.com/lheckemann/nixbuntu-samples/

**Review by philiptaron — APPROVED** [review]
I ran this and found it all worked.

```
$ nix-build --expr 'let pkgs = import ./. { }; in pkgs.callPackage ({ vmTools, ... }:
let distro = vmTools.debDistros.ubuntu2004x86_64; in
vmTools.makeImageFromDebDist {
  inherit (distro) name fullName urlPrefix packagesLists;
  packages = distro.packages ++ ["systemd" "zsh" "vim"];
}) { }'
```

File: pkgs/build-support/vm/default.nix [hunk 1]
```diff
@@ -72,8 +72,8 @@ rec {
 
     for o in $(cat /proc/cmdline); do
       case $o in
-        mountDisk=1)
-          mountDisk=1
+        mountDisk=*)
+          mountDisk=''${mountDisk#mountDisk=}
```
[the remark this answers is not in the document]
Comment (lheckemann) [reply pkgs/build-support/vm/default.nix:76]: They escape the `${` so it's not interpreted as an antiquotation by Nix. Yes, the syntax is awful :D

[the remark this answers is not in the document]
Comment (lheckemann) [reply pkgs/build-support/vm/default.nix:76]: (relevant manual section: https://nixos.org/manual/nix/stable/language/values.html?highlight=escap#primitives )

[the remark this answers is not in the document]
Comment (philiptaron) [reply pkgs/build-support/vm/default.nix:76]: Thanks. I learned something this morning!

That syntax is not great, for sure, and it's not helped that $$ (which says it makes the next $ not have any "special meaning") produces `$$` not `$`, which I expected it did.

**Pull request merged**